### ISSUE TITLE
num2date no longer supports multi-dimensional arrays

### ISSUE THREAD

**Issue by shoyer** [Unidata/netcdf4-python#354]
```
In [1]: import netCDF4

In [2]: import numpy as np

In [3]: netCDF4.num2date(np.array([[0, 1], [2, 3]]), 'days since 2000-01-01')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-34e6828dbb65> in <module>()
----> 1 netCDF4.num2date(np.array([[0, 1], [2, 3]]), 'days since 2000-01-01')

/Users/shoyer/dev/netcdf4-python/utils.pyx in netCDF4.num2date (netCDF4.c:5829)()
    240                 secs = msecsd // 1.e6
    241                 msecs = numpy.round(msecsd - secs*1.e6)
--> 242                 td = timedelta(days=days,seconds=secs,microseconds=msecs)
    243                 # add time delta to base date.
    244                 date = basedate + td

TypeError: unsupported type for timedelta microseconds component: numpy.ndarray
```

This worked on 1.1.4:
```
In [1]: import netCDF4

In [2]: import numpy as np

In [3]: netCDF4.num2date(np.array([[0, 1], [2, 3]]), 'days since 2000-01-01')
Out[3]:
array([[datetime.datetime(2000, 1, 1, 0, 0),
        datetime.datetime(2000, 1, 2, 0, 0)],
       [datetime.datetime(2000, 1, 3, 0, 0),
        datetime.datetime(2000, 1, 4, 0, 0)]], dtype=object)

In [4]: netCDF4.__version__
Out[4]: '1.1.4'
```

This is also likely related to #351.

**Comment by jswhit** [comment]
fix in pull request #358 (with test)

**Comment by jswhit** [comment]
pull request #358 merged, closing now..
